### PR TITLE
(PUP-5022) update generator skeleton to new README template

### PR DESCRIPTION
**Pull request by DavidS**

### CODE REVIEW

File: lib/puppet/module_tool/skeleton/templates/generator/README.md.erb [hunk 1]
```diff
@@ -2,8 +2,7 @@
 
 #### Table of Contents
 
-1. [Overview](#overview)
-2. [Module Description - What the module does and why it is useful](#module-description)
+1. [Description](#description)
 3. [Setup - The basics of getting started with <%= metadata.name %>](#setup)
     * [What <%= metadata.name %> affects](#what-<%= metadata.name %>-affects)
```
Comment (kylog) [inline lib/puppet/module_tool/skeleton/templates/generator/README.md.erb]: Adjust the following numbers? Not strictly required markdown-wise, but increases cognitive load to see the sequence 1,3,4,... ("is something wrong here?").

If the concern is number of lines changed, you could make them all `1.` and then never have this problem again :)

File: lib/puppet/module_tool/skeleton/templates/generator/README.md.erb [hunk 2]
```diff
@@ -13,67 +12,73 @@
...
 
-## Overview
+## Description
 
-A one-maybe-two sentence summary of what the module does/what problem it solves.
-This is your 30 second elevator pitch for your module. Consider including
-OS/Puppet version it works with.
+Start with a one- or two-sentence summary of what the module does and/or what
+problem it solves. This is your 30-second elevator pitch for your module.
+Consider including OS/Puppet version it works with.
```
Comment (kylog) [inline lib/puppet/module_tool/skeleton/templates/generator/README.md.erb]: Trailing whitespace.

File: lib/puppet/module_tool/skeleton/templates/generator/README.md.erb [hunk 3]
```diff
@@ -13,67 +12,73 @@
...
+facts, along with the parameters for each. Users refer to this section (thus
+the name "Reference") to find specific details; most users don't read it per
+se.
 
-## Limitations
+### Limitations
 
-This is where you list OS compatibility, version compatibility, etc.
+This is where you list OS compatibility, version compatibility, etc. If there
+are Known Issues, you might want to include them under their own heading here.
```
Comment (kylog) [inline lib/puppet/module_tool/skeleton/templates/generator/README.md.erb]: Add this as a `###` (or `####`?) here and flag it as `(Optional)`? This seems to be describing a Known Issues section, and elsewhere this template actually *has* the section.

File: lib/puppet/module_tool/skeleton/templates/generator/README.md.erb [hunk 4]
```diff
@@ -13,67 +12,73 @@
 5. [Limitations - OS compatibility, etc.](#limitations)
 6. [Development - Guide for contributing to the module](#development)
```
Comment (kylog) [inline lib/puppet/module_tool/skeleton/templates/generator/README.md.erb]: Reference and Limitations have become `###` below. Indent them here for consistency?

**Comment by DavidS** [comment]
Thanks for the comments. I seem to have to redo that copy&paste training ;-)

**Comment by jbondpdx** [comment]
:+1:

**Comment by joshcooper** [comment]
If this is required for 4.3.0, please retarget for stable

**Comment by DavidS** [comment]
PR'd a cherry-pick of this in #4426. Is that proper?

**Comment by joshcooper** [comment]
Thanks @DavidS. We will automatically merge the change from stable to master, so I'm going to close this PR.

**Pull request closed without merging**